Registration views

AndriiBuhalo opened the pull request:
![image](https://user-images.githubusercontent.com/87270025/151398324-b9bb4297-2bd7-4e41-a680-aa4ed59d3b31.png)
![image](https://user-images.githubusercontent.com/87270025/151398874-2eed4833-8308-4b8b-b51a-3a85fbd2b4ec.png)

AndriiBuhalo committed:
add views on registration

AndriiBuhalo committed:
add locales for registration, add edit registration, add permited param for devise (add first name... phone)

AndriiBuhalo committed:
fix bug in new registration button

Shabaldas commented on app/controllers/application_controller.rb:
```diff
@@ -14,4 +15,10 @@ def set_locale
   end
+
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up) { |u| u.permit(:first_name, :last_name, :phone, :email, :password, :password_confirmation) }
```
```u``` -is not correct name for block value

Shabaldas commented on app/controllers/application_controller.rb:
```diff
@@ -14,4 +15,10 @@ def set_locale
+  def configure_permitted_parameters
+    devise_parameter_sanitizer.permit(:sign_up) { |u| u.permit(:first_name, :last_name, :phone, :email, :password, :password_confirmation) }
+
+    devise_parameter_sanitizer.permit(:account_update) { |u| u.permit(:first_name, :last_name, :phone, :email, :password, :password_confirmation) }
```
the same

Shabaldas commented on app/javascript/controllers/password_controller.js:
```diff
@@ -0,0 +1,17 @@
+  static targets = ['unhide']
+
+  password(e) {
+    if (this.unhideTarget.type === "password") {
```
chose one type of quotes in this file

Shabaldas commented on app/views/devise/sessions/new.html.erb:
```diff
@@ -1,46 +1,44 @@
-                  <%= f.label :email, t('global.registration.email_or') %>
-                  <%= f.input :email,
+              <div class="">
+                <div class="">
```
why do you need empty class here?

Shabaldas commented on app/views/devise/sessions/new.html.erb:
```diff
@@ -1,46 +1,44 @@
                   <%= f.label :password %>
-                  <%= f.password_field :password, autocomplete: "current-password", class: 'form-control unhide', "data-modal-target": "unhide" %>
-                  <a data-action="click->modal#password" class="absolute bottom-3.5 right-2 password-field-icon-1 fas fa-eye-slash"></a>
+                  <%= f.password_field :password, autocomplete: "current-password", class: ' w-full form-control unhide', "data-password-target": "unhide" %>
```
redundant space in class

Shabaldas commented:
lets use ```"``` in all htm.erb, js files

and 

```'``` in all .rb files

so please update your html.erb files in this PR

AndriiBuhalo committed:
change name for block value and change quotes in this file

Shabaldas commented on app/views/devise/shared/_links.html.erb:
```diff
@@ -1,23 +1,25 @@
+  <%= link_to "Sign up", new_registration_path(resource_name) %><br />
 <% end %>
+
+<%- if devise_mapping.recoverable? && controller_name != 'passwords' && controller_name != 'registrations' %>
```
move this logic to helper_method

Shabaldas commented on app/views/devise/sessions/new.html.erb:
```diff
@@ -1,46 +1,44 @@
                   <%= f.label :password %>
-                  <%= f.password_field :password, autocomplete: "current-password", class: 'form-control unhide', "data-modal-target": "unhide" %>
-                  <a data-action="click->modal#password" class="absolute bottom-3.5 right-2 password-field-icon-1 fas fa-eye-slash"></a>
+                  <%= f.password_field :password, autocomplete: "current-password", class:' w-full form-control unhide', "data-password-target": "unhide" %>
```
realy?
```'``` => ```"```
and remove redundant space in class

Shabaldas commented on app/views/devise/registrations/new.html.erb:
```diff
@@ -1,25 +1,110 @@
+            </div>
+            <div class="mt-1 relative">
+              <%= f.password_field :password, autocomplete: "current-password", class: 'w-full rounded form-control unhide', "data-password-target": "unhide" %>
+              <a data-action="click->password#password" class="absolute bottom-3.5 right-2 password-field-icon-1 fas fa-eye-slash"></a>
```
why do you use ```a``` tag instead of ```div``` ?

AndriiBuhalo committed:
Fix rubocop issues

AndriiBuhalo committed:
add views on registration

 add locales for registration, add edit registration, add permited param for devise (add first name... phone)

 fix bug in new registration button

change name for block value and change quotes in this file

Fix rubocop issues

AndriiBuhalo committed:
Merge branch 'registration_views' of https://github.com/Shabaldas/storm_rails_tailwind into registration_views

AndriiBuhalo committed:
Merge branch 'master' into registration_views
